lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -658,6 +674,22 @@ instance FromJSON WalletPostData where
 instance ToJSON  WalletPostData where
     toJSON = genericToJSON defaultRecordTypeOptions
 
+instance FromJSON AccountPublicKey where
+    parseJSON bytes = do
+        xs <- parseJSON bytes
+        case convertFromBase Base16 $ T.encodeUtf8 xs of
+            Left _ -> fail "wrong account public key format"
+            Right bs ->
+                pure $ AccountPublicKey $ ApiT $ unsafeDeserializeXPub bs
```

Comment:
I am not sure if `unsafeDeserializeXPub` is ok here? The same for `serializeXPub` below

Reply:
Will it not throw (i.e "something went wrong" for the user) if provided valid hex, but which is an invalid key?

Reply:
Whola! Yes. Do not use any `unsafe` method in the API when presented user-facing inputs :scream: ! 
The "unsafe" constructors are only tolerated at the boundaries with the database, where we have controlled first what we inserted!

Reply:
it was temporary...now removed unsafe code. thanks for being vigilant here!